File: app/src/main/java/com/android/ashwiask/tvmaze/di/TvMazeDbModule.java
package com.android.ashwiask.tvmaze.di;

import android.arch.persistence.room.Room;
import android.content.Context;

import com.android.ashwiask.tvmaze.db.TvMazeDatabase;
import com.android.ashwiask.tvmaze.db.favouriteshow.ShowDao;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

@Module
public class TvMazeDbModule {
    @Singleton
    @Provides
    public TvMazeDatabase provideTvMazeDatabase(Context context) {
        return Room.databaseBuilder(context,
                TvMazeDatabase.class, TvMazeDatabase.DATABASE_NAME)
                .fallbackToDestructiveMigration()
                .allowMainThreadQueries()
                .build();
    }

    @Singleton
    @Provides
    public ShowDao provideShowDao(TvMazeDatabase tvMazeDatabase) {
        return tvMazeDatabase.showDao();
    }
}

File: app/src/main/java/com/android/tvmaze/db/TvMazeDbModule.kt
package com.android.tvmaze.db

import android.content.Context
import androidx.room.Room
import com.android.tvmaze.db.favouriteshow.ShowDao
import dagger.Module
import dagger.Provides
import dagger.hilt.InstallIn
import dagger.hilt.android.components.ApplicationComponent
import dagger.hilt.android.qualifiers.ApplicationContext

import javax.inject.Singleton

@InstallIn(ApplicationComponent::class)
@Module
object TvMazeDbModule {
    @Singleton
    @Provides
    fun provideTvMazeDatabase(@ApplicationContext context: Context): TvMazeDatabase {
        return Room.databaseBuilder(
            context,
            TvMazeDatabase::class.java, TvMazeDatabase.DATABASE_NAME
        )
            .fallbackToDestructiveMigration()
            .allowMainThreadQueries()
            .build()
    }

    @Singleton
    @Provides
    fun provideShowDao(tvMazeDatabase: TvMazeDatabase): ShowDao {
        return tvMazeDatabase.showDao()
    }
}
